xds/internal/resolver/cluster_specifier_plugin_test.go

```diff
@@ -39,13 +40,89 @@ import (
 	"google.golang.org/grpc/serviceconfig"
 	"google.golang.org/grpc/xds/internal/balancer/clustermanager"
 	"google.golang.org/grpc/xds/internal/clusterspecifier"
+	xdsresolver "google.golang.org/grpc/xds/internal/resolver"
+	protov2 "google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/anypb"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 
 	v3listenerpb "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
 	v3routepb "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
 )
 
+const (
+	defaultTestTimeout      = 10 * time.Second
+	defaultTestShortTimeout = 100 * time.Microsecond
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+func (s) TestRegister(t *testing.T) {
+	if resolver.Get(xdsresolver.Scheme) == nil {
+		t.Errorf("Scheme %q is not registered", xdsresolver.Scheme)
+	}
+}
```

Comment:
I find this test strange since I've never seen anything testing the registry from init() calls. I feel like other tests can fail as a result of this not failing which is how I've previously called this, but I don't feel strongly about it.

Reply:
Yeah, I would say it isn't very common to have this test. But I don't think it hurts to have it either.

> I feel like other tests can fail as a result of this not failing

I don't understand why/how other tests would fail as a result of this test. Other tests do read from the registry, and so, at that point, this is a redundant test for sure.